EIPS/eip-1283.md

```diff
@@ -63,7 +63,8 @@ the following logic:
       gas to refund counter.
   * If *original value* does not equal *current value* (this storage
     slot is dirty), 200 gas is deducted. Apply both of the following
-    clauses.
+    clauses. If *gasleft* is less then or equal 2300, fail the transaction
```

Comment:
I would recommend to apply the gasleft check in the beginning, before any of the branches here are applied.